diff --git a/kernel/nvidia/nv-procfs.c b/kernel/nvidia/nv-procfs.c
index 1ca106d..90865e4 100644
--- a/kernel/nvidia/nv-procfs.c
+++ b/kernel/nvidia/nv-procfs.c
@@ -360,7 +360,8 @@ nv_procfs_read_registry(
     registry_keys = ((nvl != NULL) ?
             nvl->registry_keys : nv_registry_keys);
 
-    return seq_printf(s, "Binary: \"%s\"\n", registry_keys);
+    seq_printf(s, "Binary: \"%s\"\n", registry_keys);
+    return 0;
 }
 
 static ssize_t
@@ -560,7 +561,8 @@ nv_procfs_read_text_file(
     void *v
 )
 {
-    return seq_puts(s, s->private);
+    seq_puts(s, s->private);
+    return 0;
 }
 
 NV_DEFINE_PROCFS_SINGLE_FILE(text_file);
